src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,278 @@
+package datawave.query.util;
+
+import org.apache.accumulo.core.data.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handles the serialization and deserialization of the Accumulo value in a record of the Datawave Metadata table that has a column family of "f" and
+ * a column qualifier that is prefixed with the string "compressed-" like "compressed-csv" for example. This is a class used to help compress the date and
+ * frequency values that are aggregated together to by the FrequencyTransformIterator and manipulated in the FrequencyFamilyCounter. The byte array is in
+ * regular expression format (YEAR(4BYTE-FREQUENCY){366}))* . Explained verbally a four byte representation of Year followed by 366 (Leap year) 4 byte holders
+ * for frequency. In this version all the arrays have a slot for the Feb 29 leap year frequency whether it is there or not. There aren't any delimiters between
+ * years and frequencies which adds to the compression. Each Accumulo row for this "aggregated" frequency "map" would be 10 x ( 4 + (366 x * 4) ) bytes long for
+ * a 10 year capture: 14620 bytes.
+ */
+
+public class DateFrequencyValue {
+    
+    private static final Logger log = LoggerFactory.getLogger(DateFrequencyValue.class);
+    
+    // An intermediate data structure build up by iterating through uncompressDateFrequencies
+    private HashMap<YearKey,byte[]> compressedDateFrequencies = new HashMap<>();
+    // The mapping of SimpleDateFormat strings to frequency counts. This is passed in by the
+    // FrequencyFamilyCounter object to this classes serialize function
+    private HashMap<String,Integer> uncompressedDateFrequencies = null;
```

Comment:
I think everything might work a little smoother and be more readable if this were a TreeMap<YearMonthDay, Frequency> instead of HashMap<String, Integer>.  The Frequency object could be mutable allowing aggregation of frequencies which avoid excessive Integer object allocations.  The YearMonthDay can store year and ordinal (julian date) values avoid excessive string manipulations.
```
class YearMonthDay implements Comparable<YearMonthDay> {
  int year;
  int julian;
  YearMonthDay(String value) {
     // parse value into year and ordinal
  }
  int getYear() {
     return year;
  }
  int getJulian()
     return julian;;
  }
}
```
```
class Frequency {
  int value 
  Frequency(value) {
     this.value = value;
  }
  int getValue()  {
     return value;
  }
  void addFrequency(int value) {
     this.value += value;
  }
}
```